### PR TITLE
When a single test is run, do not run others with overlapping names

### PR DESCRIPTION
**Pull request by SomeoneToIgnore**

### CODE REVIEW

File: crates/ra_ide/src/runnables.rs [hunk 1]
```diff
@@ -10,46 +10,95 @@ use ra_syntax::{
...
-    } else if fn_def.has_atom_attr("bench") {
-        RunnableKind::Bench { name: name.to_string() }
     } else {
-        return None;
+        let test_id = if let Some(module) = fn_def
+            .syntax()
+            .ancestors()
+            .find_map(ast::Module::cast)
+            .and_then(|module| source_binder.to_def(InFile::new(file_id.into(), module)))
+        {
```
Comment (flodiebold) [inline crates/ra_ide/src/runnables.rs]: I think you should be able to use `source_binder.analyze(InFile::new(file_id.into(), fn_def), None).module()`

Comment (flodiebold) [reply crates/ra_ide/src/runnables.rs]: or even simpler `source_binder.to_def(InFile::new(file_id.into(), fn_def)).module(db)`

Comment (SomeoneToIgnore) [reply crates/ra_ide/src/runnables.rs]: Great, this way it resolves the file as module also, thanks.

**Pull request merged**